### PR TITLE
Pull in PSR12

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

File: MO4/ruleset.xml [hunk 1]
```diff
@@ -140,6 +150,14 @@
     </rule>
     <!-- Forbid useless alias for classes, constants and functions -->
     <rule ref="SlevomatCodingStandard.Namespaces.UselessAlias"/>
+    <!-- Require empty newlines before and after uses -->
+    <rule ref="SlevomatCodingStandard.Namespaces.UseSpacing">
+        <properties>
+            <property name="linesCountBeforeFirstUse" value="1"/>
+            <property name="linesCountBetweenUseTypes" value="0"/>
```
Comment (mmoll) [inline MO4/ruleset.xml:157]: This didn't get support in GH-126, however, PSR-12 also wants to see it like this, so I'm trying again...

File: MO4/ruleset.xml [hunk 2]
```diff
@@ -16,6 +16,16 @@
         <exclude name="Symfony.Commenting.License"/>
     </rule>
 
+    <!-- Also include PSR12 -->
+    <rule ref="PSR12">
+        <!-- exclude sniffs that are disabled on purpose by this standard -->
+        <exclude name="Generic.Files.LineLength.TooLong"/>
+        <!-- not compatible with other rules we have later on -->
+        <exclude name="PSR12.ControlStructures.ControlStructureSpacing"/>
```
Comment (mmoll) [inline MO4/ruleset.xml:24]: Really puzzled by these multi line condition rules: https://www.php-fig.org/psr/psr-12/#51-if-elseif-else

**Comment by mmoll** [comment]
I tried to pull in as much as possible from PSR12 while minimizing conflicts. In projects that extend MO4 and have own rules, there might be some new conflicts that need to get resolved by adjusting the other rules or excluding PSR12 stuff, but that's expected.

**Comment by xalopp** [comment]
Merci @mmoll

**Pull request merged**